src/auth/views/auth/join/join.ts

```diff
@@ -34,14 +35,17 @@ export default class RouteAuthJoin extends BaseRouteComponent {
 		return this.$gettext('auth.join.page_title');
 	}
 
-	async onJoin(formModel: any) {
+	async onJoining(formModel: FormModel) {
 		// We store these so we can log them in automatically once their
 		// verification happens.
 		this.setCredentials({
```

Comment:
Get rid of setCredentials through the store - pass these in through session storage, including the auth token. This way, we can even auto login people from the app section of the site.

Reply:
Also, don't emit join event. simply do the router push in this function.